app/src/main/java/com/hedvig/app/util/SpringUtils.kt

```diff
@@ -0,0 +1,113 @@
+package com.hedvig.app.util
+
+import android.view.View
+import androidx.annotation.IdRes
+import androidx.dynamicanimation.animation.DynamicAnimation
+import androidx.dynamicanimation.animation.SpringAnimation
+import androidx.dynamicanimation.animation.SpringForce
+import com.hedvig.app.R
+
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * An extension function which creates/retrieves a [SpringAnimation] and stores it in the [View]s
+ * tag.
+ */
+fun View.spring(
+    property: DynamicAnimation.ViewProperty,
+    stiffness: Float = 500f,
+    damping: Float = SpringForce.DAMPING_RATIO_NO_BOUNCY,
+    startVelocity: Float? = null
+): SpringAnimation {
+    val key = getKey(property)
+    var springAnim = getTag(key) as? SpringAnimation?
+    if (springAnim == null) {
+        springAnim = SpringAnimation(this, property)
+        setTag(key, springAnim)
+    }
+    springAnim.spring = (springAnim.spring ?: SpringForce()).apply {
+        this.dampingRatio = damping
+        this.stiffness = stiffness
+    }
+    startVelocity?.let { springAnim.setStartVelocity(it) }
+    return springAnim
+}
+
+/**
+ * Map from a [DynamicAnimation.ViewProperty] to an `id` suitable to use as a [View] tag.
+ */
+@IdRes
+private fun getKey(property: DynamicAnimation.ViewProperty): Int {
+    return when (property) {
+        SpringAnimation.TRANSLATION_X -> R.id.translation_x
```

Comment:
Strange that we need to do it like this feels like this should be supported in a better way from the system

Reply:
This is actually the thing that Nick Butcher created and showed off in his talk: https://www.youtube.com/watch?v=f3Lm8iOr4mE

I had written one that was heavily inspired by that one, but I found the actual implementation in the source code for Plaid and decided to use that one straight away.

It looks a bit weird but it is indeed the way to go to implement this kind of Spring animation extension 🙂